src/ark/utils/deepcell_service_utils.py

```diff
@@ -99,85 +175,49 @@ def create_deepcell_output(deepcell_input_dir, deepcell_output_dir, fovs=None,
+                # extract segmentation masks from deepcell output
+                extract_deepcell_response(deepcell_output_dir, fov_group, batch_num, wc_suffix,
+                                          nuc_suffix)
 
-        # pass the zip file to deepcell.org
-        print('Uploading files to DeepCell server.')
-        status = run_deepcell_direct(
-            zip_path, deepcell_output_dir, host, job_type, scale, timeout
-        )
+            total_time = time.time() - start
 
-        # ensure execution is halted if run_deepcell_direct returned non-zero exit code
         if status != 0:
-            print("The following FOVs could not be processed: %s" % ','.join(fov_group))
-            return
-
-        # extract the .tif output
-        print("Extracting tif files from DeepCell response.")
-        zip_names = io_utils.list_files(deepcell_output_dir, substrs=[".zip"])
-
-        zip_files = [os.path.join(deepcell_output_dir, name) for name in zip_names]
-
-        # sort by newest added
-        zip_files.sort(key=os.path.getmtime)
-
-        with ZipFile(zip_files[-1], "r") as zipObj:
-            for name in zipObj.namelist():
-                # this files will only ever be suffixed with feature_0.tiff or feature_1.tiff
-                if '_feature_0.tif' in name:
-                    resuffixed_name = name.replace('_feature_0', wc_suffix)
-                else:
-                    resuffixed_name = name.replace('_feature_1', nuc_suffix)
+            unprocessed_fovs[batch_num] = fov_group
+        if total_time >= timeout:
+            print(f"This batch exceeded the allotted processing time of {timeout / 60} minutes "
+                  f"and will be skipped.")
 
-                mask_path = os.path.join(deepcell_output_dir, resuffixed_name)
-
-                # DeepCell uses .tif extension, append extra f to account for .tiff standard
-                mask_path += 'f'
-
-                # read the file from the .zip file and save as segmentation mask
-                byte_repr = zipObj.read(name)
-                ranked_segmentation_mask = (_convert_deepcell_seg_masks(byte_repr)).squeeze()
-                image_utils.save_image(mask_path, ranked_segmentation_mask)
-
-            # verify that all the files were extracted
-            for fov in fov_group:
-                if fov + '_feature_0.tif' not in zipObj.namelist():
-                    warnings.warn(f'Deep Cell whole cell output file was not found for {fov}.')
-                if fov + '_feature_1.tif' not in zipObj.namelist():
-                    warnings.warn(f'Deep Cell nuclear output file was not found for {fov}.')
-
-    # make calls in parallel
-    if parallel:
-        with ThreadPoolExecutor() as executor:
-            executor.map(_zip_run_extract, fov_groups, range(len(fov_groups)))
-            executor.shutdown(wait=True)
-    else:
-        list(map(_zip_run_extract, fov_groups, range(len(fov_groups))))
+    if unprocessed_fovs:
```

Comment:
In the case that a batch fails the first time but passes the next, would this not still print the warning?

There should be a check under `if status == 0` to account for this:

```
if status == 0:
    # extract segmentation masks from deepcell output
    extract_deepcell_response(deepcell_output_dir, fov_group, batch_num, wc_suffix, nuc_suffix)
    
    # since the batch_num is now processed, it should be removed from the dict
    if batch_num in unprocessed_fovs:
        unprocessed_fovs.pop(batch_num)

    # addresses the previous comment
    break

Reply:
Kinda hard to see with the code change structure, but I moved the if statement out of the loop so it only add the batch to `unprocessed_fovs` after the loop concludes (either bc of a successful output zip or timeout). Easier to see it here: https://github.com/angelolab/ark-analysis/blob/f40a37d11eb46453fdc753c2a8c8383ee279b76e/src/ark/utils/deepcell_service_utils.py#L207